### PR TITLE
Append last n to first linked list

### PR DESCRIPTION
**Pull request by anshika272**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
 given a singly linked list of integers along with an integer 'N'. Write a function to append the last 'N' nodes towards the front of the singly linked list and returns the new head to the list, Using Cpp.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
 issue #5200

### CODE REVIEW

**Comment by anshika272** [comment]
@HarshCasper  kindly please review the code.

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/* 
+"Copyright [2021] <Anshika Dubey>"
+You have been given a singly linked list of integers along with an integer 'N'. Write a function to append the last 'N' nodes towards the front of the singly linked list and returns the new head to the list.
```
Comment (bhav09) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: The description can be cut down to multiple lines. 
After every full stop, bring the sentence to the next line. Improves readablity.

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+2
+1 2 3 4 5 -1
+3
+10 20 30 40 50 60 -1
+5
+Sample output 1-
+3 4 5 1 2
+20 30 40 50 60 10
+
+Time Complexity-
```
Comment (bhav09) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Complexity has to be above Sample I/O

**Review by bhav09 — CHANGES_REQUESTED** [review]
Minor changes need to be done. Rest fine.
Also, comment on your code a little more, it lacks that.

**Comment by anshika272** [comment]
@bhav09 please review the changes

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Use comments above the line of code not beside it

**Comment by anshika272** [comment]
@HarshCasper  @bhav09  please review the changes

**Comment by anshika272** [comment]
@HarshCasper @bhav09 @ankitaggarwal23 please review the changes.

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/* 
+"Copyright [2021] <Anshika Dubey>"
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: remove the copyright comment line

File: C-Plus-Plus/README.md [hunk 4]
```diff
@@ -200,9 +200,11 @@
 - [Check if two given trees are Quasi Isomorphic or not](ds/Quasi_Isomorphic_Binary_Trees.cpp)
 - [Build a Balanced BST with given vector of values](ds/Balanced_BST_From_a_vector_of_values.cpp)
 - [Given a BST check if it is AVL Tree or not](ds/AVL_Tree_or_not.cpp)
+- [AppendLastNToFirst](ds/AppendLastNToFirst - Linked List.cpp)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/README.md]: Rewrite it and also rename your file name as well 

`- [Append Last N To First](ds/Append_Last_N_To_First.cpp)`

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+You have been given a singly linked list of integers along with an integer 'N'.
+Write a function to append the last 'N' nodes towards the front of the singly linked list and returns the new head to the list.
+
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+// create a class node
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Add a new line above comments

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+class Node {
+public:
+	int data;
+	Node *next;
+	Node(int data) {
+		this->data = data;
+		this->next = NULL;
+	}
+};
+// taking input from the user for linked list
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Add a new line above comments

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+			tail = newnode;
+		} else {
+			tail->next = newnode;
+			tail = newnode;
+		}
+		cin >> data;
+	}
+	return head;
+}
+//printing the linked list if head is given
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Add a new line above comments

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+//printing the linked list if head is given 
+void print(Node *head) {
+	Node *temp = head;
+	while (temp != NULL) {
+		cout << temp->data << " ";
+		temp = temp->next;
+	}
+	cout << endl;
+}
+//finding the length of the linked list
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Add a new line above comments

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+int length1(Node *head) {
+    // Write your code here
+    if(head == NULL)
+        return 0;
+    if(head->next == NULL)
+        return 1;
+    Node* temp = head;
+    return 1 + length1(temp->next);
+}
+// function to append last n elements of linked list at front
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Add a new line above comments

File: C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+	// taking input for number of test cases
+	int t;                            
+	cin >> t;
+	while (t--) {
+		Node *head = takeinput();
+		int n;
+		cin >> n;
+		head = appendLastNToFirst(head, n);
+		print(head);
+	}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/AppendLastNToFirst - Linked List.cpp]: Remove the test case part as we don't have online judge.  Use proper display message how you are reading and writing outputs

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**